### PR TITLE
feat(aria-label): new rule to flag incorrect use of aria-label, aria-labelledby, and aria-describeby

### PR DESCRIPTION
**Pull request by straker**
Closes issue: #1575

### CODE REVIEW

File: lib/rules/aria-label.json [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{
+  "id": "aria-label",
+  "selector": "[aria-label], [aria-labelledby], [aria-describedby]",
+  "tags": ["cat.aria", "best-practice"],
+  "metadata": {
+    "description": "Ensure aria-label(ledby) is only used on non-static elements",
+    "help": "Use aria-label, aria-labelledby, and aria-describedby on elements with non-static roles"
```
Comment (WilcoFiers) [inline lib/rules/aria-label.json:7]: I'd like to see some test data on this. What is the problem we are solving with this rule?

**Review by WilcoFiers — CHANGES_REQUESTED** [review]
We're going to need to run this rule by the JC. Can you do some testing for this to show what the problem is that comes from this? I suspect we're going to get complaints about this one.

**Comment by straker** [comment]
I think this PR is not needed and we should instead use the [Prohibited States and Props from ARIA 1.2](https://www.w3.org/TR/wai-aria-1.2/#prohibitedattributes) as they cover this exact use case but in a more generic way.

Related discussions:
- https://github.com/w3c/aria/issues/756
- https://github.com/w3c/aria/pull/985

**Comment by straker** [comment]
Note, this may not work with the `<i>` attribute as it doesn't have a role mapping. I will do some testing `aria-label` on different elements to see how screen readers handle it.

**Comment by straker** [comment]
From Scott O'Hara on asking about HTML elements with no role mapping:

> no mapping = generic
> generic doesn't allow naming

**Comment by straker** [comment]
Tested the following DOM in different browsers:

```html
<div aria-label="my label">foo</div>
<span aria-label="my label">foo</span>
<i aria-label="my label">foo</i>
<p aria-label="my label">foo</p>

<div aria-labelledby="label">foo</div>
<span aria-labelledby="label">foo</span>
<i aria-labelledby="label">foo</i>
<p aria-labelledby="label">foo</p>
```

No browser / screen reader combination read the `aria-label` or the `aria-labelledby` attributes. All names returned as "foo"

**Pull request closed without merging**